### DISCUSSION TITLE
PageFormat

### DISCUSSION THREAD

**Discussion by gerardoscoppetta** [opensheetmusicdisplay/opensheetmusicdisplay#1546] [Q&A] [answered]
HI.
I can only display the page format in A4 or A3 in both portrait and landscape versions but I can't try with custom dimensions as explained here
https://opensheetmusicdisplay.github.io/classdoc/classes/PageFormat.html#constructor
I tried a bit but I get this error
Type error :t.match in not a function.
Can you see some well written examples? Maybe I'm the one doing something wrong.
Thank you

**Comment by sschmidTU** [comment]
Where exactly does the error occur and what are you using as input?
match is not used in the PageFormat constructor.
The error occurs because match() is called on something that is not a string. match is a function on strings.

**Reply by gerardoscoppetta** [reply]
HI.
here.
https://www.suonosubito.net/test_opensheetmusicdisplay/index007.php
you can see in the console.
If I remove pageFormat or set it to A4 it works but if I leave it like this I get that error.
<img width="1073" alt="Screenshot 2024-05-13 alle 18 10 35" src="https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/assets/41266915/951bb178-d386-4688-9932-1bf73ea4199e">
<img width="350" alt="Screenshot 2024-05-13 alle 18 11 23" src="https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/assets/41266915/66dc9df0-dd4c-4dda-87e3-1ccb88e2b061">

**Reply by sschmidTU** [reply, marked as answer]
pageFormat in options expects a string, not width+height.
https://opensheetmusicdisplay.github.io/classdoc/interfaces/IOSMDOptions.html#pageFormat
If you want to use a custom format, set `osmd.EngravingRules.PageFormat` as a `PageFormat` object directly (use the PageFormat constructor).

**Reply by gerardoscoppetta** [reply]
OK thank you.
